extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ParquetFileReader.java

```diff
@@ -27,27 +29,53 @@ public class ParquetFileReader {
     private static final int FOOTER_LENGTH_SIZE = 4;
     private static final String MAGIC_STR = "PAR1";
     static final byte[] MAGIC = MAGIC_STR.getBytes(StandardCharsets.US_ASCII);
+    public static final String S3_PARQUET_FILE_URI_SCHEME = "s3";
+    private static final String S3_PARQUET_FILE_URI_PREFIX = "s3://";
 
     public final FileMetaData fileMetaData;
     private final SeekableChannelsProvider channelsProvider;
     private final Path rootPath;
     private final MessageType type;
 
+    // TODO Where should I keep it?
+    public static URI convertToURI(final String filePath) {
+        // We need to encode spaces in the path string that are not allowed in a URI
+        if (filePath.startsWith(S3_PARQUET_FILE_URI_PREFIX)) {
+            try {
+                return new URI(filePath.replace(" ", "%20"));
+            } catch (final URISyntaxException e) {
+                throw new UncheckedDeephavenException("Failed to convert file path " + filePath + " to URI", e);
+            }
+        } else {
+            return Path.of(filePath).toUri();
+        }
+    }
+
     public ParquetFileReader(final String filePath, final SeekableChannelsProvider channelsProvider)
             throws IOException {
-        this.channelsProvider = channelsProvider;
-        // Root path should be this file if a single file, else the parent directory for a metadata
-        // file
-        rootPath =
-                filePath.endsWith(".parquet") ? Paths.get(filePath) : Paths.get(filePath).getParent();
+        this(convertToURI(filePath), channelsProvider);
+    }
 
+    public ParquetFileReader(final URI parquetFileURI, final SeekableChannelsProvider channelsProvider)
+            throws IOException {
+        this.channelsProvider = channelsProvider;
+        final Path filePath;
+        if (parquetFileURI.getScheme() != null && parquetFileURI.getScheme().equals(S3_PARQUET_FILE_URI_SCHEME)) {
```

Comment:
When one side may be null, I often invert it to make it simpler: `S3_PARQUET_FILE_URI_SCHEME.equals(parquetFileURI.getScheme())`.